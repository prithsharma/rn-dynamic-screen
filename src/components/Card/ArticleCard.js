import React from 'react';
import {
  Image,
  TouchableWithoutFeedback,
  Text,
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';


let styles;

function Card(props) {
  const {
    onPress,
    titleText,
    imgUri,
    descriptionText,
    // timestamp,
  } = props;
  const timestampText = 'About 30 mins ago';

  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.container}>
        <View style={styles.headerRow}>
          <Image
            style={styles.image}
            source={{ uri: imgUri }}
          />
          <View style={styles.headerTitleCol}>
            <Text style={styles.titleText}>
              {titleText}
            </Text>
            <Text style={styles.timestamp}>
              {timestampText}
            </Text>
          </View>
        </View>
        <View style={styles.descriptionContainer}>
          <Text style={styles.descriptionText} numberOfLines={2}>
            {unescape(descriptionText)}
          </Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
}

Card.propTypes = {
  onPress: PropTypes.func,
  titleText: PropTypes.string.isRequired,
  imgUri: PropTypes.string.isRequired,
  descriptionText: PropTypes.string,
};


Card.defaultProps = {
  onPress: Function.prototype,
  descriptionText: '',
};

styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    padding: 6,
    paddingVertical: 8,
    elevation: 2,
    borderRadius: 1,
    shadowColor: 'black',
    shadowOffset: { width: 0.5, height: 1 },
    shadowOpacity: 0.4,
    shadowRadius: 2,
    marginHorizontal: 5,
  },
  headerRow: {
    flexDirection: 'row',
    marginBottom: 4,
  },
  headerTitleCol: {
    flex: 3,
  },
  titleText: {
    flex: 1,
    fontSize: 16,
    fontWeight: 'bold',
  },
  image: {
    flex: 2,
    marginRight: 8,
    resizeMode: 'contain',
    aspectRatio: 3 / 2,
  },
  descriptionText: {
    fontSize: 12,
  },
  timestamp: {
    color: 'grey',
    fontSize: 10,
    marginBottom: 4,
  },
});

export default Card;
