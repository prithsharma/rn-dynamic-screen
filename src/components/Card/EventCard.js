import React from 'react';
import {
  Image,
  TouchableHighlight,
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

let styles;

function Card(props) {
  const {
    onPress,
    imgUri,
  } = props;

  return (
    <TouchableHighlight onPress={onPress}>
      <View style={styles.container}>
        <Image
          style={styles.image}
          source={{ uri: imgUri }}
        />
        <View style={styles.descriptionContainer} />
      </View>
    </TouchableHighlight>
  );
}

Card.propTypes = {
  onPress: PropTypes.func,
  imgUri: PropTypes.string.isRequired,
};


Card.defaultProps = {
  onPress: Function.prototype,
};

styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    padding: 10,
  },
  image: {
    width: '100%',
    height: undefined,
    aspectRatio: 150 / 76,
    resizeMode: 'contain',
    marginTop: 8,
    marginBottom: 4,
  },
});

export default Card;
