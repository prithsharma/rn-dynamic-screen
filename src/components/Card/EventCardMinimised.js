import React from 'react';
import {
  TouchableWithoutFeedback,
  Text,
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';
import Countdown from '../Countdown';


let styles;

function Card(props) {
  const {
    onPress,
    city,
    localTime,
    title,
    timestampStr,
    showCountdown,
  } = props;
  const detailsStr = `${city}, ${localTime}`;

  const countdown = showCountdown && (
    <View style={styles.minimisedCountdown}>
      <Countdown targetTimestamp={timestampStr} scale={0.7} />
    </View>
  );

  return (
    <TouchableWithoutFeedback
      onPress={onPress}
      style={styles.touchableContainer}
    >
      <View style={styles.container}>
        <View style={styles.details}>
          <Text style={styles.titleText}>{title}</Text>
          <Text style={styles.detailsText}>{detailsStr}</Text>
        </View>
        {countdown}
      </View>
    </TouchableWithoutFeedback>
  );
}

Card.propTypes = {
  onPress: PropTypes.func,
  city: PropTypes.string,
  localTime: PropTypes.string,
  title: PropTypes.string,
  timestampStr: PropTypes.string.isRequired,
  showCountdown: PropTypes.bool.isRequired,
};

Card.defaultProps = {
  onPress: Function.prototype,
  city: '',
  localTime: '',
  title: '',
};

styles = StyleSheet.create({
  touchableContainer: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
  },
  container: {
    borderTopWidth: 2,
    borderTopColor: '#ffcc01',
    backgroundColor: 'black',
    flexDirection: 'row',
    padding: 6,
  },
  details: {
    // alignItems: 'flex-start',
    paddingLeft: 8,
    justifyContent: 'center',
    flex: 1,
  },
  minimisedCountdown: {
    color: 'white',
  },
  titleText: {
    fontSize: 15,
    color: '#ffcc01',
    fontWeight: 'bold',
  },
  detailsText: {
    fontSize: 12,
    color: 'white',
  },
});

export default Card;
