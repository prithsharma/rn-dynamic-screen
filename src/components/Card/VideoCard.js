import React from 'react';
import {
  Image,
  TouchableWithoutFeedback,
  Text,
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';
import moment from 'moment';


let styles;

function Card(props) {
  const {
    onPress,
    titleText,
    imgUri,
    descriptionText,
    timestampStr,
  } = props;
  const timestampText = moment(timestampStr).fromNow();

  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.container}>
        <View>
          <Text style={styles.titleText}>
            {titleText}
          </Text>
          <Image
            style={styles.image}
            source={{ uri: imgUri }}
          />
          <View style={styles.descriptionContainer}>
            <Text style={styles.timestamp}>
              {timestampText}
            </Text>
            <Text style={styles.descriptionText} numberOfLines={2}>
              {descriptionText}
            </Text>
          </View>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
}

Card.propTypes = {
  onPress: PropTypes.func,
  titleText: PropTypes.string.isRequired,
  imgUri: PropTypes.string.isRequired,
  descriptionText: PropTypes.string,
  timestampStr: PropTypes.string.isRequired,
};


Card.defaultProps = {
  onPress: Function.prototype,
  descriptionText: '',
};

styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    padding: 10,
    elevation: 2,
    borderRadius: 1,
    shadowColor: 'black',
    shadowOffset: { width: 0.5, height: 1 },
    shadowOpacity: 0.4,
    shadowRadius: 2,
    marginHorizontal: 5,
  },
  titleText: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  image: {
    width: '100%',
    marginTop: 8,
    marginBottom: 4,
    resizeMode: 'contain',
    aspectRatio: 16 / 9,
  },
  descriptionText: {
    fontSize: 12,
  },
  timestamp: {
    color: 'grey',
    fontSize: 10,
    marginBottom: 4,
  },
});

export default Card;
