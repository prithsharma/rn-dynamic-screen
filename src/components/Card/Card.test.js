/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow } from 'react-native-testing-library';
import ArticleCard from './ArticleCard';
import EventCard from './EventCard';
import EventCardMinimised from './EventCardMinimised';
import VideoCard from './VideoCard';

it('ArticleCard renders correctly', () => {
  expect(shallow(
    <ArticleCard
      titleText="Regian Eersel's game"
      imgUri="https://facebook.github.io/react-native/docs/assets/favicon.png"
      timestampStr="2018-05-11T07:30:00+00:00"
      descriptionText="Regian Eersel's game"
    />,
  ).output).toMatchSnapshot();
});

it('EventCard renders correctly', () => {
  expect(shallow(
    <EventCard
      imgUri="http://res.cloudinary.com/onefc/image/upload/v1554878097/events/Bangkok-2019-05-10/upcoming"
      timestampStr="2018-05-11T07:30:00+00:00"
      showCountdown={false}
    />,
  ).output).toMatchSnapshot();
});

it('EventCardMinimised renders correctly', () => {
  expect(shallow(
    <EventCardMinimised
      timestampStr="2018-05-11T07:30:00+00:00"
      showCountdown={false}
    />,
  ).output).toMatchSnapshot();
});

it('VideoCard renders correctly', () => {
  expect(shallow(
    <VideoCard
      titleText="Regian Eersel's game"
      timestampStr="2018-05-11T07:30:00+00:00"
      imgUri="http://res.cloudinary.com/onefc/image/upload/v1554878097/events/Bangkok-2019-05-10/upcoming"
    />,
  ).output).toMatchSnapshot();
});
