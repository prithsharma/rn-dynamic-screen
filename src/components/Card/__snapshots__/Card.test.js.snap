// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 1,
        "elevation": 2,
        "marginHorizontal": 5,
        "padding": 6,
        "paddingVertical": 8,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 0.5,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://facebook.github.io/react-native/docs/assets/favicon.png",
          }
        }
        style={
          Object {
            "aspectRatio": 1.5,
            "flex": 2,
            "marginRight": 8,
            "resizeMode": "contain",
          }
        }
      />
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Regian Eersel's game
        </Text>
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 10,
            }
          }
        >
          a year ago
        </Text>
      </View>
    </View>
    <View>
      <Text
        numberOfLines={2}
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Regian Eersel's game
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`EventCard renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "marginBottom": 10,
      }
    }
  >
    <ScalableImage
      background={false}
      onSize={[Function]}
      source={
        Object {
          "uri": "http://res.cloudinary.com/onefc/image/upload/v1554878097/events/Bangkok-2019-05-10/upcoming",
        }
      }
      width={750}
    />
  </View>
</TouchableWithoutFeedback>
`;

exports[`EventCardMinimised renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderTopColor": "#ffcc01",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "marginBottom": 10,
        "padding": 6,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffcc01",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 12,
          }
        }
      >
        , 
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`VideoCard renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 1,
        "elevation": 2,
        "marginHorizontal": 5,
        "padding": 10,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 0.5,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 2,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Regian Eersel's game
      </Text>
      <Image
        source={
          Object {
            "uri": "http://res.cloudinary.com/onefc/image/upload/v1554878097/events/Bangkok-2019-05-10/upcoming",
          }
        }
        style={
          Object {
            "aspectRatio": 1.7777777777777777,
            "marginBottom": 4,
            "marginTop": 8,
            "resizeMode": "contain",
            "width": "100%",
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 10,
              "marginBottom": 4,
            }
          }
        >
          a year ago
        </Text>
        <Text
          numberOfLines={2}
          style={
            Object {
              "fontSize": 12,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</TouchableWithoutFeedback>
`;
